projects/04-burger-queen/README.md

```diff
@@ -490,3 +491,22 @@ yarn start
 * [Qué es Serverless? | FooBar - YouTube](https://www.youtube.com/watch?v=_SYHUpLi-2U)
 * [Firebase](https://firebase.google.com/)
 * [Serverless Architectures - Martin Fowler](https://www.martinfowler.com/articles/serverless.html)
+
+## 7. Funcionalidades para reforzar OA de promesas
+
+Te sugerimos implementar las siguientes funcionalidades para que puedas reforzar los objetivos de aprendizaje de promesas vistos en el proyecto de MD Links.
+
+* Agrega la opción de actualizar 2 o más pedidos que esten en la cocina. Muestra al usuario un **único** mensaje de confirmación cuando **todos** los pedidos hayan sido actualizados con éxito. Recuerda, firestore al [actualizar un documento](https://firebase.google.com/docs/firestore/manage-data/add-data#update-data) retorna una promesa. ¿Cómo podrías mostrar el mensaje de confirmación unicamente cuando todas las promesas se hayan resuelto? Te sugerimos revisar la función [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all).
+* Cuando un pedido finalice se debe hacer una petición HTTP a la url [https://salty-escarpment-74924.herokuapp.com/](https://salty-escarpment-74924.herokuapp.com/) que responderá de forma aleatoria si el cliente tiene el 50% de descuento en su cuenta. Para esto deberás llamar la función _getDiscount_ mostrada a continuación. Termina la implementación de la función según los comentarios que hay en ella. Te sugerimos revisar la documentacion de [HTTP Codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status),  [fetch](https://developer.mozilla.org/en-US/docs/Web/API/fetch) y [creación de promesas](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/Promise).
+
+  ```js
+  export const getDiscount = () => {
+    return new Promise((resolve, reject) => {
+
+      //1. Realiza una peticion HTTP a la URL https://salty-escarpment-74924.herokuapp.com/
+      //2. Si la peticion retorna codigo 200 entonces resuelve la promesa
```

Comment:
Acabo de ver que la API devuelve un código de status 500 cuando no hay descuento:
<img width="809" alt="api-discount" src="https://user-images.githubusercontent.com/810823/159568372-22a5f07b-6d22-4272-838f-943af77358c1.png">

Un 500 debe devolverse cuando hay un error en el servidor, y este no es el caso, yo sugiero que cuando no haya descuento, devuelva 200 pero tenga un parámetro en el body de la respuesta:
```
{
  "discount": false,
  "msg": "Esta vez no tienes descuento. Inténtalo en tu próxima visita"
}
```

Reply:
Tienes razon @santiaguf. En los dos casos deberia responder 200 y variar la llave gotDiscount. Lo que queria hacer era que la estudiante lideara con multiples codigos http. ¿Es muy enredado dejar que el servicio responda 200 con gotDiscount en true y false pero que de forma aleatoria tambien falle con 500?

Reply:
Creo que para eso podrías usar una respuesta 404 (descuento no encontrado).
tal cómo [mencionan en esta guía](https://r.bluethl.net/how-to-design-better-apis#heading-7-use-reasonable-http-status-codes), es mejor evitar los códigos 500.

Reply:
Mira https://github.com/ssinuco/buger-queen-discount/pull/1